client/lib/promote-post/index.ts

```diff
@@ -328,10 +329,11 @@ export const requestDSPHandleErrors = async < T >(
 	siteId: number,
 	apiUri: string,
 	method: SupportedDSPMethods = 'GET',
-	body: Record< string, unknown > | undefined = undefined
+	body: Record< string, unknown > | undefined = undefined,
+	apiVersion = 1
```

Comment:
i think it is cleared to use 
apiVersion: string = 'v1'

passing a number like 1.1 might have unexpected results

Reply:
Ah I actually thought the opposite, I originally made it as a string and thought a number would be easier to validate?

I'm already using 1.1 in[ the code above](https://github.com/Automattic/wp-calypso/pull/96346/files#diff-48799417c950d435b194ceafd46e75b18f31ea45fe1d33ba1f2247aca343ca7cR179) and is working as expected.

But I'm not super opinionated, I can make it a string 👍

Reply:
i think it reads better also for developers 
having a `'v1.1'` is more clear than having a number like 1.1

Reply:
for validation we could do something like that

`type SupportedApiVersions = 'v1' | 'v1.1';`

`apiVersion: SupportedApiVersions = 'v1'`